controllers/release/adapter_test.go

```diff
@@ -804,6 +805,55 @@ var _ = Describe("Release adapter", Ordered, func() {
 		})
 	})
 
+	When("EnsureReleaseExpirationTimeIsAdded is called", func() {
+		var adapter *adapter
+		var expectedExpirationTime *metav1.Time
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+			newReleasePlan := &v1alpha1.ReleasePlan{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "release-plan",
+					Namespace: "default",
+				},
+				Spec: v1alpha1.ReleasePlanSpec{
+					Application:            application.Name,
+					ReleaseGracePeriodDays: metadata.DefaultReleaseGracePeriodDays,
+					Target:                 "default",
+				},
+			}
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlanContextKey,
+					Resource:   newReleasePlan,
+				},
+			})
+
+			expireDays := time.Duration(metadata.DefaultReleaseGracePeriodDays)
+			creationTime := adapter.release.CreationTimestamp
+			expectedExpirationTime = &metav1.Time{Time: creationTime.Add(time.Hour * 24 * expireDays)}
+		})
+
+		It("should set the ExpirationTime and continue", func() {
+			result, err := adapter.EnsureReleaseExpirationTimeIsAdded()
+			Expect(adapter.release.Status.ExpirationTime).To(Equal(expectedExpirationTime))
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should set continue if the ExpirationTime is already set", func() {
```

Comment:
I think this test would be better if the release.Status.ExpirationTime is set to something else at the start (line 849) instead. The way it is written, you manually set it to the default, then call the operation which sets it to the default anyways, so you can't really be sure if it was overwritten

Reply:
fixed.